### PR TITLE
Enable handling of 'where' expressions and fix deprecations

### LINKED ISSUE #445: [Julia 0.6] Wrong category for UnionAll types; Constant instead of Type
```julia
"""
This is a Blob
"""
struct Blob{T} end
```
On 0.5:
image

On 0.6:
image

### PR DESCRIPTION
**Pull request by helgee**
x-ref #445

### CODE REVIEW

File: src/Utilities/Utilities.jl [hunk 1]
```diff
@@ -300,6 +300,9 @@ doccat(b::Binding, ::Type)  = "Method"
 
 doccat(::Function) = "Function"
 doccat(::DataType) = "Type"
+if VERSION >= v"0.6.0-dev.2122"
```
Comment (mortenpi) [inline src/Utilities/Utilities.jl]: Not sure actually which is better practice, but `isdefined(Base, :UnionAll)` should work here as well, right?

**Comment by helgee** [comment]
I couldn't resist to also fix the low-hanging deprecations.

**Comment by helgee** [comment]
There is only the following deprecation warning left:
```
WARNING: deprecated syntax "inner constructor Diff(...) around /Users/helge/.julia/v0.6/Documenter/src/Utilities/TextDiff.jl:71".
Use "Diff{T}(...) where T" instead.
```
But I am not sure how to handle this one.

**Comment by helgee** [comment]
I fixed the last deprecation warning.

**Comment by helgee** [comment]
BTW I also added the requested tests. Anything left to do?

File: REQUIRE [hunk 2]
```diff
@@ -1,3 +1,3 @@
 julia 0.5
-Compat 0.8
+Compat 0.18
```
Comment (tkelman) [inline REQUIRE]: doesn't look like any of the type aliases are parametric, so 0.17 should be enough

**Comment by helgee** [comment]
Fixed.

**Comment by mortenpi** [comment]
Looks good, thanks for doing the 0.6 stuff as well. One last request -- could you squash your commits into two, one for the deprecations and one for the `where` thing?

**Comment by helgee** [comment]
Done 👍

**Comment by mortenpi** [comment]
Thanks!

**Pull request merged**